cmd/common/image_functions.go

```diff
@@ -198,95 +304,203 @@ func GetUntaggedManifests(ctx context.Context, acrClient api.AcrCLIClientInterfa
 			return nil, err
 		}
 	}
-	// Remove all manifests that should not be deleted
-	for i := 0; i < len(candidates); i++ {
-		if !ignoreList.Contains(*candidates[i].Digest) {
-			// if a manifest has no tags, is not part of a manifest list and can be deleted then it is added to the
-			// manifestsForCommand array.
-			if *(candidates[i].ChangeableAttributes).DeleteEnabled && *(candidates[i].ChangeableAttributes).WriteEnabled {
-				manifestsForCommand = append(manifestsForCommand, *candidates[i].Digest)
-				if dryRun && !ignoreReferrerManifests {
-					fmt.Printf("%s/%s@%s\n", loginURL, repoName, *candidates[i].Digest)
-				}
-			}
+
+	// Wait for all the goroutines to finish or return an error if one of them failed
+	if err := group.Wait(); err != nil {
+		return nil, err
+	}
+
+	// Remove everything from the candidates list that is in the ignoreList
+	for _, manifest := range candidates {
+		if _, ok := ignoreList.Load(*manifest.Digest); ok {
+			// Remove the manifest from the candidates list
+			delete(candidates, *manifest.Digest)
 		}
 	}
 
-	return &manifestsForCommand, nil
+	for _, manifest := range candidates {
+		// Add the manifest to the list of manifests to delete
+		manifestsToDelete = append(manifestsToDelete, *manifest.Digest)
+	}
+
+	return manifestsToDelete, nil
```

Comment:
When something is added to the ignore list, can we output why it's added to the ignore list, I think there might be some confusion customers might have been running into because we're not very specific about why something is being ignored. So from their point of view they're like "hey it's untagged, why isn't it deleted" so here we should say something like "digest: blah is a sub-manifest of manifest: blah, adding to ignore list" or something along those lines